sw/device/sca/ecc256_keygen_serial.c

```diff
@@ -144,6 +198,77 @@ static void p256_run_keygen(uint32_t mode, const uint32_t *seed,
   sca_set_trigger_low();
 }
 
+static void ecc256_ecdsa_secret_keygen_batch(const uint8_t *data,
+                                             size_t data_len) {
+  uint32_t num_traces = 0;
+  uint32_t out[kEcc256SeedNumWords];
+  uint32_t batch_digest[kEcc256SeedNumWords];
+  uint8_t dummy[kEcc256SeedNumBytes];
+  SS_CHECK(data_len == sizeof(num_traces));
+  num_traces = read_32(data);
+
+  if (num_traces > kNumBatchOpsMax) {
+    LOG_ERROR("Too many traces for one batch.");
+    return;
+  }
+
+  // zero the batch digest
+  for (uint32_t j = 0; j < kEcc256SeedNumWords; ++j) {
+    batch_digest[j] = 0;
+  }
+
+  for (uint32_t i = 0; i < num_traces; ++i) {
+    if (run_fixed) {
+      memcpy(batch_seeds[i], ecc256_seed, kEcc256SeedNumBytes);
+    } else {
+      // One additional PRNG run required to be in sync with host.
+      // Necessary because the key-plaintex-generator on the host generates
+      // two random arrays at once, while only the secon is used (ktp.next()[1])
+      prng_rand_bytes(dummy, kEcc256SeedNumBytes);
+      prng_rand_bytes((unsigned char *)batch_seeds[i], kEcc256SeedNumBytes);
+    }
+    if (en_masks) {
+      // One additional PRNG run required to be in sync with host.
+      // Necessary because the key-plaintex-generator on the host generates
+      // two random arrays at once, while only the secon is used (ktp.next()[1])
+      prng_rand_bytes((unsigned char *)batch_masks[i], kEcc256SeedNumBytes);
+      prng_rand_bytes((unsigned char *)batch_masks[i], kEcc256SeedNumBytes);
```

Comment:
I think the first call of `prng_rand_bytes((unsigned char *)batch_masks[i], kEcc256SeedNumBytes)` should be `prng_rand_bytes(dummy, kEcc256SeedNumBytes)`, shouldn't it?
It would be good to be consistent throughout the code. Maybe it is even better, to solve the root cause of the sync problem. See my other comment.